cmake/Platform/Common/MSVC/Configurations_msvc.cmake

```diff
@@ -8,39 +8,22 @@
 include(cmake/Platform/Common/Configurations_common.cmake)
 include(cmake/Platform/Common/VisualStudio_common.cmake)
 
-set(LY_MSVC_SUPPORTED_GENERATORS
-    "Visual Studio 15"
-    "Visual Studio 16"
-)
-set(FOUND_SUPPORTED_GENERATOR)
-foreach(supported_generator ${LY_MSVC_SUPPORTED_GENERATORS})
-    if(CMAKE_GENERATOR MATCHES ${supported_generator})
-        set(FOUND_SUPPORTED_GENERATOR TRUE)
-        break()
-    endif()
-endforeach()
-# VS2017's checks since it defaults the toolchain and target architecture to x86
-if(CMAKE_GENERATOR MATCHES "Visual Studio 15")
-    if(CMAKE_VS_PLATFORM_NAME AND CMAKE_VS_PLATFORM_NAME STREQUAL "Win32") # VS2017 has Win32 as the default architecture
-        message(FATAL_ERROR "Win32 architecture not supported, specify \"-A x64\" when invoking cmake")
-    endif()
-    if(NOT CMAKE_VS_PLATFORM_TOOLSET_HOST_ARCHITECTURE STREQUAL "x64") # There is at least one library (EditorLib) that make the x86 linker to run out of memory
-        message(FATAL_ERROR "x86 toolset not supported, specify \"-T host=x64\" when invoking cmake")
-    endif()
+if(NOT CMAKE_GENERATOR MATCHES "Visual Studio 1[6-7]")
+    message(FATAL_ERROR "Generator ${CMAKE_GENERATOR} not supported")
+endif()
+
+# For the other cases, verify that it wasn't invoked with an unsupported architecture. defaults to x86 architecture
```

Comment:
The comment should be updated to mention that the default is x64

Reply:
And can also remove the SUPPORTED_VS_PLATFORM_NAME_OVERRIDE handling since that is not a CMake variable and nothing overrides that